spring-cloud-commons/src/main/java/org/springframework/cloud/util/InetUtils.java

```diff
@@ -26,6 +26,103 @@
 @CommonsLog
 public class InetUtils {
 
+	private final ExecutorService executorService;
+	private final InetUtilsProperties properties;
+
+	public InetUtils(final InetUtilsProperties properties) {
+		this.properties = properties;
+		this.executorService = Executors
```

Comment:
feel free to not address this now, but this will eagerly create a thread, even if never used. This is a way out, if interested: https://github.com/square/okhttp/blob/master/okhttp/src/main/java/okhttp3/internal/framed/FramedConnection.java#L156